### PR TITLE
sap_hypervisor_node_preconfigure: Added OCP Auth, replaced oc wait with k8s native function

### PR DESCRIPTION
**Pull request by newkit**
- Added authentication via kubeconfig, kubeadmin & password. Need to provide ca-cert too in order to validate connection.
- The oc wait commands have been replaced with a k8s native function.
- Do not download virtctl to the jumphost anylonger.
- Housekeeping: removed unused kargs.yml.

This includes a better version as used in this [PR](https://github.com/sap-linuxlab/community.sap_infrastructure/pull/7) and therefore makes it obsolete.

### CODE REVIEW

**Comment by newkit** [comment]
@0xFelix could you please review? Sorry for the extra loop.

File: roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/install-sriov-operator.yml [hunk 1]
```diff
@@ -40,15 +40,12 @@
   ansible.builtin.pause:
     minutes: 3
 
-- name: Copy patch to enable unsupported NICs
-  ansible.builtin.copy:
-    src: sriov-enabled-unsupported-nics.sh
-    dest: "{{ __sap_hypervisor_node_preconfigure_register_tmpdir.path }}/sriov-enabled-unsupported-nics.sh"
-    mode: "0755"
+- name: Enable unsupported NICs for SRIOV usage
```
Comment (sean-freeman) [inline roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/install-sriov-operator.yml:43]: Typo, `SR-IOV`

**Review by sean-freeman — CHANGES_REQUESTED** [review]

File: playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml [hunk 2]
```diff
@@ -1,19 +1,30 @@
 ---
-
 - name: Ansible Play to run sap_hypervisor_node_preconfigure Ansible Role
   hosts: all
   gather_facts: true
   serial: 1
-
   vars:
     sap_hypervisor_node_platform: redhat_ocp_virt
```
[the remark this answers is not in the document]
Comment (newkit) [reply playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml:7]: @sean-freeman I am confused, I see `sap_hypervisor_node_preconfigure_platform` nowhere defined, only  `sap_hypervisor_node_platform`. Do you mean to replace all occurencies of `sap_hypervisor_node_platform` with `sap_hypervisor_node_preconfigure_platform`? Please elaborate.

File: playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
-    - name: Include sap_hypervisor_node_preconfigure Ansible Role
-      ansible.builtin.include_role:
-        name: sap_hypervisor_node_preconfigure
+    - name: Log into Red Hat OpenShift cluster (obtain access token)
+      community.okd.openshift_auth:
+        host: "{{ sap_hypervisor_node_preconfigure_ocp_endpoint }}"
+        username: "{{ sap_hypervisor_node_preconfigure_ocp_admin_username }}"
+        password: "{{ sap_hypervisor_node_preconfigure_ocp_admin_password }}"
+        ca_cert: "{{ sap_hypervisor_node_preconfigure_ocp_ca_cert }}"
+      register: __sap_vm_provision_register_ocp_auth_results
```
[the remark this answers is not in the document]
Comment (newkit) [reply playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml:15]: By space you mean newline? 

Login and receiving API Key is not optional but required on OCPv.

File: requirements.yml [hunk 4]
```diff
@@ -35,3 +35,10 @@ collections:
   - name: cloud.common
     type: galaxy
     version: 3.0.0
+# For OpenShift
+  - name: community.okd
+    type: galaxy
+    version: 3.0.1
+  - name: community.general
```
[the remark this answers is not in the document]
Comment (newkit) [reply requirements.yml:42]: `community.okd` and  therefore `community.general` is required by the ` community.okd.openshift_auth` call in playbook `playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml`.

File: playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sean-freeman) [reply playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml:7]: Yes, var should be amended to the correct prefix

File: playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sean-freeman) [reply playbooks/sample-sap-hypervisor-redhat-ocp-virt-preconfigure.yml:15]: Yes, newline.

Both username/password and API Key are required?  That seems highly ineffective/inefficient, will not question further though

File: requirements.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (sean-freeman) [reply requirements.yml:42]: `community.okd.openshift_auth` has a dependency on `community.general`? That's not pretty but I'll consider this closed in that case

File: roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/install-sriov-operator.yml (hunk 1 above)
Comment (newkit) [reply roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/install-sriov-operator.yml:43]: Ack, will be fixed.

File: roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/configure-worker-node.yml [hunk 5]
```diff
@@ -12,8 +7,12 @@
...
-# How to wait for node to be scheduleable? (NodeSchedulable)
 - name: Wait for all k8s nodes to be ready
-  ansible.builtin.command: oc wait --for=condition=Ready nodes --all --timeout=3600s
-  register: __sap_hypervisor_node_preconfigure_register_nodes_ready
-  changed_when: __sap_hypervisor_node_preconfigure_register_nodes_ready.rc != 0
+  kubernetes.core.k8s_info:
+    kind: Node
+    wait: yes
+    wait_condition:
+      status: True
```
[the remark this answers is not in the document]
Comment (newkit) [reply roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/configure-worker-node.yml:15]: Ack, will be fixed.

File: roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/configure-worker-node.yml [hunk 6]
```diff
@@ -12,8 +7,12 @@
...
   when: __sap_hypervisor_node_preconfigure_register_worker.networks is defined
 
-# How to wait for node to be scheduleable? (NodeSchedulable)
 - name: Wait for all k8s nodes to be ready
-  ansible.builtin.command: oc wait --for=condition=Ready nodes --all --timeout=3600s
-  register: __sap_hypervisor_node_preconfigure_register_nodes_ready
-  changed_when: __sap_hypervisor_node_preconfigure_register_nodes_ready.rc != 0
+  kubernetes.core.k8s_info:
+    kind: Node
+    wait: yes
```
[the remark this answers is not in the document]
Comment (newkit) [reply roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/configure-worker-node.yml:13]: Ack, will be fixed.

**Comment by newkit** [comment]
I believe I've resolved all the mentioned issues, could you please take a look @sean-freeman ?

**Review by rhmk — APPROVED** [review]
The requested changes were made and they LGTM

**Pull request merged**